### DISCUSSION TITLE
Appropriate Location for Interfaces Shared Between Backend and Frontend

### DISCUSSION THREAD

**Discussion by bysxx** [vercel/next.js#52484] [App Router]
Where is the appropriate folder to place interfaces used in both backend (app/api) and frontend (app/*)?

Currently, they are specified in the app/common/interfaces folder. Is there a more suitable location?

**Comment by icyJoseph** [comment]
If you wanna borrow inspiration from Next.js itself, you can always have a `shared` folder. 

The main idea is to avoid, as much as possible, mixing code that's environment dependent, like a module with a function that uses `fs` for example, should not be imported, even indirectly by a component that'll run on the client. 

Since, I think, you are asking about TypeScript interfaces, I guess the above is not an issue.

I'd say though, put them in a sibling folder to the app directory, not everything has to go in there.

**Reply by bysxx** [reply]
Thank you for your answer.

I'm currently creating a full stack template through Next.js.

Your answer was also very helpful, but I would like to hear your opinion a little more.

I would appreciate it if you could look at the repository and give me your opinion.

https://github.com/bysxx/next13-ts-template-fullstack

**Reply by icyJoseph** [reply]
Since you seem to want to place most things inside `app`, you might benefit from the `Private folder` convention, https://nextjs.org/docs/app/building-your-application/routing/colocation#private-folders, and have it as `_common`, so that those are not accidentally routed. 

This is why `common` as a sibling to `app`, is kind of ok, and maybe for organization, you could opt in to the `src` folder. 

I also don't think you need `experimental.appDir` in the next config file anymore.

**Reply by bysxx** [reply]
Thank you !!
